File: HW/HW6/main.py
def threeSum(self, nums: list[int]) -> list[list[int]]:
    array = []
    for i in range(len(nums) - 2):
        for j in range(i+1, len(nums) - 1):
            for k in range(j+1, len(nums)):
                if nums[i] + nums[j] + nums[k] == 0:
                    triplet = [nums[i], nums[j], nums[k]]
                    triplet.sort()
                    if triplet not in array:
                        array.append(triplet)
    return array


numStr = input("Type in array of integers separated by a ', ' to find triplets that sum to 0: \n")
numArrayStr = numStr.split(', ')
numbers = []
for num in numArrayStr:
    numbers.append(int(num))
triplets = threeSum(threeSum, numbers)
print(triplets)

File: HW/HW6/Main.java
import java.util.*;

public class Main {
    public List<List<Integer>> threeSum(int[] nums) {
        List<List<Integer>> array = new ArrayList<List<Integer>>(nums.length);
        for(int i = 0; i < nums.length - 2; i++) {
            for(int j = i+1; j < nums.length - 1; j++) {
                for(int k = j+1; k < nums.length; k++) {
                    if(nums[i] + nums[j] + nums[k] == 0) {
                        List<Integer> triplet = new ArrayList<>(3);
                        triplet.add(nums[i]);
                        triplet.add(nums[j]);
                        triplet.add(nums[k]);
                        Collections.sort(triplet);
                        if(!array.contains(triplet))
                            array.add(triplet);
                    }
                }
            }
        }
        return array;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        Main obj = new Main();

        System.out.print("Enter length of Array: ");
        int length = in.nextInt();
        int[] numArray = new int[length];

        for(int i = 0; i < length; i++) {
            System.out.print("Enter an integer of the Array: ");
            int num = in.nextInt();
            numArray[i] = num;
        }

        List<List<Integer>> triplets;
        triplets =  obj.threeSum(numArray);
        System.out.print(triplets.toString());
    }
}